File: INF.04-25.01/INF.04-01-25.01-SG/konsolowa/cpp/main.cpp
#include <iostream>
#include <random>
#include <numeric>

class ArrayOperations {
private:
    int *numbers;
    int size;

public:
    explicit ArrayOperations(const int size) {
        this->size = size;
        this->numbers = new int[size];

        std::default_random_engine generator;
        std::uniform_int_distribution<int> distribution(1, 1000);
        for (int i = 0; i < size; i++) {
            this->numbers[i] = distribution(generator);
        }
    }

    ~ArrayOperations() {
        delete[] this->numbers;
    }

    /**
     * nazwa metody: display
     * opis metody: metoda wyświetla wszystkie liczby w tablicy `numbers` w osobnych liniach
     * parametry: brak
     * zwracany typ i opis: brak
     * autor: 123456789112
     */
    void display() const {
        for (int i = 0; i < this->size; i++) {
            std::cout << this->numbers[i] << std::endl;
        }
    }

    [[nodiscard]] int findFirst(const int needle) const {
        for (int i = 0; i < this->size; i++) {
            if (this->numbers[i] == needle) {
                return i;
            }
        }
        return -1;
    }

    [[nodiscard]] int displayOdds() const {
        int counter = 0;
        for (int i = 0; i < this->size; i++) {
            if (this->numbers[i] % 2 == 1) {
                counter++;
                std::cout << this->numbers[i] << std::endl;
            }
        }

        return counter;
    }

    [[nodiscard]] double getMean() const {
        int total = std::accumulate(this->numbers, this->numbers + this->size, 0);
        return static_cast<double>(total) / static_cast<double>(this->size);
    }
};

int main() {
    auto arrayOps = ArrayOperations(21);

    std::cout << "Wszystkie elementy tablicy:" << std::endl;
    arrayOps.display();

    int index_42 = arrayOps.findFirst(42);
    if (index_42 != -1) {
        std::cout << "Liczbę 42 znaleziono na indeksie " << index_42 << "." << std::endl;
    }

    std::cout << "Liczby nieparzyste:" << std::endl;
    auto odds = arrayOps.displayOdds();
    std::cout << "Liczba liczb nieparzystych w tablicy: " << odds << "." << std::endl;

    std::cout << "Średnia wszystkich liczb w tablicy: " << arrayOps.getMean() << std::endl;

    return 0;
}

File: INF.04-25.01/INF.04-01-25.01-SG/konsolowa/java/main.java
import java.util.Random;

class ArrayOperations {
    private final int[] numbers;
    private final int length;

    ArrayOperations(int length) {
        this.length = length;
        this.numbers = new int[length];

        var rng = new Random();

        for (int i = 0; i < length; i++) {
            this.numbers[i] = rng.nextInt(1, 1001); // [1, 1001)
        }
    }

    /**
     * ******************
     * nazwa metody: printArray
     * opis metody: wypisuje na konsolę wszystkie liczby w tablicy
     * parametry: brak
     * zwracany typ i opis: brak
     * autor: 123456789112
     * ******************
     */
    public void printArray() {
        for (int i = 0; i < length; i++) {
            System.out.println(i + ": " + this.numbers[i]);
        }
    }

    public int findFirst(int needle) {
        for (int i = 0; i < length; i++) {
            if (this.numbers[i] == needle) {
                return i;
            }
        }

        return -1;
    }

    public int printOdd() {
        int counter = 0;
        for (int i = 0; i < length; i++) {
            if (this.numbers[i] % 2 == 1) {
                counter++;
                System.out.println(this.numbers[i]);
            }
        }

        return counter;
    }

    public double average() {
        int sum = 0;
        for (int i = 0; i < length; i++) {
            sum += this.numbers[i];
        }

        return (double) sum / this.length;
    }
}

class Main {
    public static void main(String[] args) {
        var arrayOps = new ArrayOperations(21);

        System.out.println("Wszystkie elementy tablicy:");
        arrayOps.printArray();

        var index_42 = arrayOps.findFirst(42);
        if (index_42 != -1) {
            System.out.println("Liczbę 42 znaleziono na indeksie " + index_42 + ".");
        }

        System.out.println("Liczby nieparzyste w tablicy:");
        var numOdds = arrayOps.printOdd();
        System.out.println("Liczba liczb nieparzystych w tablicy: " + numOdds);

        var avg = arrayOps.average();
        System.out.println("Średnia wartość liczb w tablicy: " + avg);
    }
}